core/src/main/java/org/elasticsearch/search/internal/SearchContext.java

```diff
@@ -204,6 +203,17 @@ public final boolean nowInMillisUsed() {
 
     public abstract ContextIndexSearcher searcher();
 
+    /**
+     * Executes a search and cleans up releasable data structures after the search has been executed.
+     */
+    public void search(Query query, Collector collector) throws IOException {
```

Comment:
actually there are not that many callers of that method, so maybe it would be better to just have a finally block that calls clearReleasables on the call sites. It would avoid having a dedicated method for search but not eg. explain?